scripts/merge-channel-files.js

```diff
@@ -0,0 +1,71 @@
+// @ts-check
+
+const yaml = require('js-yaml');
+const fs = require('fs');
+const path = require('path');
+
+if (process.argv.includes('--help') || process.argv.includes('-h')) {
+  console.log(
+    `Usage:
+merge-channel-files.js [FLAG]...
+
+Merge the "channel update info files" used by electron-updater.
+
+Flags:
+      --channel <name>     The name of the update channel.
+  -h, --help               Print help for the script
+      --input <path>       The path of the folder that contains the files to merge.
+`
+  );
+  process.exit(0);
+}
+
+const channelFlagIndex = process.argv.indexOf('--channel');
+
+let channel;
+if (channelFlagIndex > -1) {
+  channel = process.argv[channelFlagIndex + 1];
+} else {
+  console.error('Missing required --channel flag');
+  process.exit(1);
+}
+
+const inputFlagIndex = process.argv.indexOf('--input');
+
+let channelFilesFolder;
+if (inputFlagIndex > -1) {
+  channelFilesFolder = process.argv[inputFlagIndex + 1];
+} else {
+  console.error('Missing required --input flag');
+  process.exit(1);
+}
```

Comment:
```suggestion
const channelFlagIndex = process.argv.indexOf('--channel');
if (channelFlagIndex < 0) {
  console.error('Missing required --channel flag');
  process.exit(1);
}
const channel = process.argv[channelFlagIndex + 1];
if (!channel) {
  console.error('--channel value must be set');
  process.exit(1);
}

const inputFlagIndex = process.argv.indexOf('--input');
if (inputFlagIndex < 0) {
  console.error('Missing required --input flag');
  process.exit(1);
}
const channelFilesFolder = process.argv[inputFlagIndex + 1];
if (!channelFilesFolder) {
  console.error('--input value must be set');
  process.exit(1);
}
```

It's great you've taken care of argument validation—just a proposal if you want to handle invalid values, too. Otherwise, `node ./scripts/merge-channel-files.js --channel --input` could fail.


```
node:internal/validators:162
    throw new ERR_INVALID_ARG_TYPE(name, 'string', value);
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at new NodeError (node:internal/errors:405:5)
    at validateString (node:internal/validators:162:11)
    at Object.join (node:path:1171:7)
    at Object.<anonymous> (/Users/a.kitta/dev/git/arduino-ide/scripts/merge-channel-files.js:45:33)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

I am OK if you leave it as is.

Reply:
Thanks!